### PR TITLE
Move `dtr` extension to a core extension

### PR DESCRIPTION
**Pull request by jbants**
**Related Issue(s):** #
https://github.com/radiantearth/stac-spec/pull/636

**Proposed Changes:**

1. Remove `dtr` prefix.
2. Move `dtr` to core extension.

**PR Checklist:**

- [ ] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: extensions/datetime-range/README.md [hunk 1]
```diff
@@ -11,12 +13,12 @@ An extension to provide datetime ranges with a start and an end datetime stamp i
 
 | Field Name         | Type   | Description                                                  |
 | ------------------ | ------ | ------------------------------------------------------------ |
-| dtr:start_datetime | string | **REQUIRED.** The first or start date and time for the item, in UTC. It is formatted as `date-time` according to [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6). |
-| dtr:end_datetime   | string | **REQUIRED.** The last or end date and time for the item, in UTC. It is formatted as `date-time` according to [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6). |
+| start_datetime | string | **REQUIRED.** The first or start date and time for the item, in UTC. It is formatted as `date-time` according to [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6). |
```
Comment (m-mohr) [inline extensions/datetime-range/README.md]: It doesn't tell anything how this relates to the datetime field in the Item spec.

File: extensions/datetime-range/README.md [hunk 2]
```diff
@@ -11,12 +13,12 @@ An extension to provide datetime ranges with a start and an end datetime stamp i
...
+| end_datetime   | string | **REQUIRED.** The last or end date and time for the item, in UTC. It is formatted as `date-time` according to [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6). |
 
 **Note:** All STAC Items that use this extension MUST include both fields, which enables a user to
 search STAC records by the provided times.
 
 ## Implementations
 
-None yet, still in proposal stage.
+3 proprietary implementations
```
Comment (m-mohr) [inline extensions/datetime-range/README.md]: Do we have a public one to link to?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
+- [datetime-range extension](extensions/datetime-range/README.md): Removed extension prefix from example and schema. Moved extension to core.
```
Comment (m-mohr) [inline CHANGELOG.md]: What does "moving to core" mean?

File: extensions/datetime-range/examples/example-video.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
-  "stac_version": "0.8.0",
+  "stac_version": "0.8.1",
   "stac_extensions": [
-    "datetime-range"
+    "dtr"
```
Comment (m-mohr) [inline extensions/datetime-range/examples/example-video.json]: This is wrong, the stac_extensions are the folder names an that's still datetime-range.

File: extensions/datetime-range/README.md (hunk 2 above)
Comment (jbants) [reply extensions/datetime-range/README.md]: Not that i'm aware of. Sentinel 1 STAC supposedly uses it, but its not available as a public catalog yet

File: extensions/datetime-range/examples/example-video.json (hunk 4 above)
Comment (jbants) [reply extensions/datetime-range/examples/example-video.json]: fixed

File: extensions/datetime-range/examples/example-video.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-  "stac_version": "0.8.1",
+  "stac_version": "0.8.0",
```
Comment (m-mohr) [inline extensions/datetime-range/examples/example-video.json]: Why has that changed?

Comment (jbants) [reply extensions/datetime-range/examples/example-video.json]: accidental delete :) fixed

File: extensions/datetime-range/README.md [hunk 6]
```diff
@@ -1,8 +1,11 @@
 # Datetime Range Extension Specification (`dtr`)
```
Comment (m-mohr) [inline extensions/datetime-range/README.md]: There's still the extension prefix in the header. Replace `dtr` with `-`

Comment (jbants) [reply extensions/datetime-range/README.md]: Ahh. Good catch. Fixed

**Pull request merged**